src/pudl/helpers.py

```diff
@@ -1513,7 +1513,9 @@ def weighted_average(
         A table with ``by`` columns as the index and the weighted ``data_col``.
     """
     df["_data_times_weight"] = df[data_col] * df[weight_col]
-    df["_weight_where_notnull"] = df.loc[df[data_col].notnull(), weight_col]
+    df.loc[df[data_col].notnull(), "_weight_where_notnull"] = df.loc[
+        df[data_col].notnull(), weight_col
+    ]
```

Comment:
i am honestly not sure what changed that would have tipped this assignment quirk over into an error but I can understand why assigning a column as a portion of that column could cause a problem and this fixed the error.